### PR TITLE
Depend on Swift Collections

### LINKED ISSUE #82: Un-vendor the `swift-collections` implementation
`multipart-kit` currently contains a somewhat outdated version of `OrderedSet` and `OrderedDictionary` taken from the then-current pre-release version of `swift-collections`; this vendored implementation should be replaced with a dependency on `swift-collections` 1.x.

cc @siemensikkema

### PR DESCRIPTION
**Pull request by ahmdyasser**
Removes the vended copy of Swift Collections and adds it as a dependency.

Resolves #82

### CODE REVIEW

**Comment by siemensikkema** [comment]
I had an old PR for this https://github.com/vapor/multipart-kit/pull/74 that is now a bit out of date but comparing that with this one I see that we should also adjust or delete the NOTICE.md file. Are there perhaps other references or comments that should be updated?

**Comment by siemensikkema** [comment]
By the way, I think the reason that the other PR was never merged was because at the time we didn't want to bump the minimum Swift requirement but that is now irrelevant since Swift NIO requires 5.4. Maybe our Package.swift files should be updated to communicate that as well? @gwynne?

**Comment by ahmdyasser** [comment]
> I had an old PR for this #74 that is now a bit out of date but comparing that with this one I see that we should also adjust or delete the NOTICE.md file. Are there perhaps other references or comments that should be updated?

Thanks for the tip!, I think I will delete the [Attribution](https://github.com/vapor/multipart-kit#attribution) part from README.md file

**Comment by 0xTim** [comment]
I'll have a look at this in the next couple of days. We'll need to update Vapor first to bump the swift tools version so this can be pulled in. You'll also need to update CI as well but that might need to be done by @gwynne

**Comment by ahmdyasser** [comment]
@0xTim The test failed because I forgot to import the `Collections` library to some files. How can I test this PR again?

**Comment by 0xTim** [comment]
@ahmdyasser each commit needs approving until you've had your first PR merged unfortunately

**Comment by ahmdyasser** [comment]
@0xTim  So should I do something or wait for someone to fix it?

**Comment by 0xTim** [comment]
@ahmdyasser you can see the latest run. You need to update the manifest to bump the swift tools version to 5.4. I need to update the central CI matrix

**Comment by ahmdyasser** [comment]
@0xTim  Hi Tim! there are failed test cases, should I do something to fix it or you will do it?

**Comment by 0xTim** [comment]
@ahmdyasser It's working now. Will be a bit before we merge though because I need to update all of Vapor's dependents, then Vapor and then we can merge

File: README.md [hunk 1]
```diff
@@ -133,5 +133,3 @@ false
 Note that the array elements always include the index (as opposed to just `[]`) in order to support complex nesting.
 
 ### Attribution
```
Comment (0xTim) [inline README.md:135]: @ahmdyasser Can you remove this title as it doesn't point to anything now? Thanks!

**Review by 0xTim — CHANGES_REQUESTED** [review]
One small README update then good to merge

**Comment by ahmdyasser** [comment]
> One small README update then good to merge

done.

**Review by 0xTim — APPROVED** [review]
Thanks!

**Comment by 0xTim** [comment]
Original comment:

> - add swift-collections as a dependency
> - delete OrderedCollections folder

**Pull request merged**